t/02-bowels/90-verifier.t

```diff
@@ -0,0 +1,384 @@
+    $conf_file->remove if $trusted; # left over from prior failed run?
+
+    if (!$trusted) {
+        # Module::Signature is going to sign using your default key and there is no
+        # mechanism to override that, so temporarily make a GPG configuration
+        # change
+
+        my $fh = $conf_file->opena or die "Could not open gpg.conf for write";
+        print $fh "default-key 90D594AF\n";
+        undef $fh;
+    }
+
+    # Do the signing shuffle
+    Module::Signature::sign();
+
+    $conf_file->remove; # lets be clean
+
+    if ($damage) {
+        # Invalidate the signature by adding an extra file
+        $dir->file('FOO')->touch();
+        my $fh = $dir->file('MANIFEST')->opena
+          or die "Could not open MANIFEST for append";
+        print $fh "\nFOO";
+        undef $fh;
+    }
+
+
+    # Rebuild the archive
+    # TODO there has got to be a more portable way to do this
+    # Maybe Archive::Any::Create, but thats a lot of work
+
+    chdir $dir->parent;
+    if ( $archive =~ /\.zip$/ ) {
+        system( 'zip', '-r', $archive, $dir->basename ) == 0
+          or die "Failed to create new zip archive: $!";
+    }
+    else {
+        system( 'tar', '-czf', $archive, $dir->basename ) == 0
+          or die "Failed to create new tar archive: $!";
+    }
+
+    return;
+}
+
+#-----------------------------------------------------------------------------
+
+sub test_verify {
+
+    my ($level, $local ) = @_;
+
+    #-----------------------------------------------------------------------------
+
+    subtest "Distribution is good" => sub {
+
+        my $dist = $local->get_distribution(
+            author => 'GOOD',
+            archive => 'Foo-1.2.tar.gz',
+        );
+
+        my $verifier = Pinto::Verifier->new(
+            local    => $dist->native_path,
+            upstream => $dist->source,
+            level    => $level,
+        );
+
+        ok($verifier->verify_upstream(), "Upstream verifies")
+            or warn $verifier->failure, "\n";
+        ok($verifier->verify_local(),    "Local verifies")
+            or warn $verifier->failure, "\n";
+    };
+
+    #-----------------------------------------------------------------------------
+
+    subtest "Distribution is bad" => sub {
+
+        my $dist = $local->get_distribution(
+            author  => 'BAD',
+            archive => 'Bar-1.2.tar.gz',
+        );
+
+        my $verifier = Pinto::Verifier->new(
+            local    => $dist->native_path,
+            upstream => $dist->source,
+            level    => $level,
+        );
+
+        ok(!$verifier->verify_upstream(), "Upstream does not verify");
+        ok(!$verifier->verify_local(),    "Local does not verify");
+    };
+}
+
+#-----------------------------------------------------------------------------
+
+subtest "Verification level 1" => sub {
+
+    # At level 1, we verify the distributions checksum using the upstream
+    # CHECKSUMS file.
+
+    my $upstream = Pinto::Tester->new();
```

Comment:
Don't worry about the stuff you've already done, but for future code, my policy is: no unnecessary parens.

So `Pinto::Tester->new;` is preferred.